### PR TITLE
[JENKINS-46831] Test behaviour when agent defined globally and on stage

### PR DESCRIPTION
**Pull request by alecharp**
* JENKINS issue(s):
    * n/a
* Description:
    * I faced a problem when the agent on the global level is a Docker container and when the agent on the stage is a normal agent. This ends up with a very odd issue. I'd like to make sure it's coming from my end and not from the plugin.
* Users/aliases to notify:
    * @abayer

### CODE REVIEW

File: pipeline-model-definition/src/test/resources/agentDockerGlobalThenLabel.groovy [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            image 'maven:3.5.0-jdk-8-alpine'
+            label 'docker'
+        }
+    }
+    stages {
+        stage("Maven") {
+            steps { sh 'echo "first agent = $WHICH_AGENT"' }
+        }
+        stage("Agent") {
+            agent { label 'some-label' }
```
Comment (abayer) [inline pipeline-model-definition/src/test/resources/agentDockerGlobalThenLabel.groovy]: This should be `other-docker` or else everything hangs. =)

**Comment by abayer** [comment]
So in a local test, even after I changed `some-label` to `other-docker` to un-hang things, I got a failure due to workspace whackiness, with a mention of [JENKINS-33510](https://issues.jenkins-ci.org/browse/JENKINS-33510). I think the problem is due to the fake agents created by `JenkinsRule.createOnlineSlave()` using local paths for everything, but not sure.

**Comment by abayer** [comment]
Nope, it's a general problem - reproduced it by hand. It's specifically that you can't invoke `sh`/`bat`/`ps` inside a `node` block within a `docker.image(...).inside` block due to [JENKINS-33510](https://issues.jenkins-ci.org/browse/JENKINS-33510). Huh!

**Comment by JonnyPower** [comment]
I'm seeing similar behaviour and was writing a similar test - I believe the container id from the outer agent is expected on the inner agent which makes no sense (the container ids are not going to be the same on different nodes)

[JENKINS-33510](https://issues.jenkins-ci.org/browse/JENKINS-33510) is supposedly resolved, but the test here still fails - and my own experience still sees the issue this test replicates

**Pull request merged**